packages/@aws-cdk/aws-rds/test/instance.test.ts

```diff
@@ -717,9 +782,71 @@ describe('instance', () => {
     instance.addRotationSingleUser();
 
     // THEN
-    expect(() => instance.addRotationSingleUser()).toThrow(/A single user rotation was already added to this instance/);
+    expect(() => {
+      instance.addRotationSingleUser({ secret: instance.secret });
+    }).toThrow(/A single user rotation for master secret was already added to this instance/);
+  });
+
+  test('add single user rotation for two additional secrets', () => {
+    const instance = new rds.DatabaseInstance(stack, 'Database', {
+      engine: rds.DatabaseInstanceEngine.SQL_SERVER_EE,
+      instanceType: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+      credentials: rds.Credentials.fromUsername('syscdk'),
+      vpc,
+    });
+    const secret1 = new rds.DatabaseSecret(stack, 'MySecret1', { username: 'user1' });
+    secret1.attach(instance);
+    const secret2 = new rds.DatabaseSecret(stack, 'MySecret2', { username: 'user2' });
+    secret2.attach(instance);
 
+    // WHEN
+    instance.addRotationSingleUser({ secret: secret1 });
+    instance.addRotationSingleUser({ secret: secret2 });
 
+    // THEN
+    expect(stack).toHaveResource('AWS::SecretsManager::RotationSchedule', {
+      SecretId: {
+        Ref: 'MySecret1E18D875B',
+      },
+      RotationLambdaARN: {
+        'Fn::GetAtt': [
+          'MySecret1RotationSingleUser7D84B21E',
+          'Outputs.RotationLambdaARN',
+        ],
+      },
+    });
+    expect(stack).toHaveResource('AWS::SecretsManager::RotationSchedule', {
+      SecretId: {
+        Ref: 'MySecret225E18D70',
+      },
+      RotationLambdaARN: {
+        'Fn::GetAtt': [
+          'MySecret2RotationSingleUser2903C638',
```

Comment:
Same comment here - we should only have a single Lambda for these (ideally).

Reply:
Ditto.